### ISSUE TITLE
[Bug]: Ubispft Connect does not update to BETA

### ISSUE THREAD

**Issue by gegy1** [Whisky-App/Whisky#1004]
### Description

After installing ubisoft connect it updates to a new version but this version seems to be the old ubisoft connect version nit the new BETA version. I need to update to the BETA version because online games with other people resulting in a version conflict. Like Anno 1800 tells the bug LR05 which means that the clients have a different version. I can verify that the other client hast the BETA version of ubisoft connect.
I can not change to BETA version in the UI of ubisoft connect also.
I tried to switch to Windows 11 but this does not solve the problem.

### Steps to reproduce

Make a new bottle
Install dodnet48 using winetricks
Install ubisoftconnect using winetricks or download it from the website
Run ubsisoft connect

After login you get the old version of ubisoft connect

<img width="730" alt="Bildschirmfoto 2024-05-31 um 01 08 22" src="https://github.com/Whisky-App/Whisky/assets/75700881/fd6b904a-ddd8-4514-b46d-0a5c2e9438e3">

### Expected behaviour

Should update/download  to the new (BETA) version of ubisoft connect.

### Logs

```bash
No logs
```

### What version of Whisky are you using?

2.3.2

### What version of macOS are you using?

Sonoma (macOS 14)

### Issue Language

- [X] Yes my issue is written in English

**Comment by gegy1** [comment]
I did another test yesterday: I installed the ubisoft client on a Windows 10 PC. After starting the client, the new BETA Ui is shown. So i thought: Why not to copy the whole ubisoft folder (Program Files x86) to the whisky bottle. So i deleted the ubisoft folder in the whisky bottle and replaced it with the folder from the windows 10 pc.
Sadly, this does not solve the issue. I guess a explicit winetrick is needed, that ubisoft client shows the BETA Ui and Anno detects the BETA client version.

**Comment by gegy1** [comment]
Seems that this is an issue on wine and Whisky using an very old version of wine.
Ubisoft Connect seems to detect an old windows version. Should be fixed in a newer wine version.
https://bugs.winehq.org/show_bug.cgi?id=49006

**Comment by gegy1** [comment]
@Gcenx I did some research meanwhile and figured out: You can set the bottle to windows 10/11 using winetricks (settings). This seems to work and let ubisoft connect start the "beta" version of the client. But there is another issue now. After showing the launchscreen of ubisoft connect you getting a big empty (transparent) screen of ubisoft connect. 
It seems the launcher is working but something is missing to display the window correctly.
I also tryed your wineskin app, as you have a newer wine engine in your wineskin launcher and get the same issue. 
When i set the bottle to Windows 7 (also using winetricks) ubisoft connect starts the old version. 
But why i need the new beta version you ask: Because Anno 1800 seems to verify something for multiplayer. I can launch the game but i am getting an "incompatible version" error LR05. My network does not have an issue, because when i am playing from my old windows pc everything works.

Do you have any idea what the bottle is missing because of the blank/transparent window for the ubisoft connect client?

My Mac is an Mini M2 with 8gb Ram

**Comment by gegy1** [comment]
@Gcenx  _I don't own any games that require Ubisoft Connect, I just know that currently it works when setting the wineprefix to Windows 7._

No need for a game to test this just install ubisoft connect from their website, set the bottle to windows 10/11 (using winetricks) and you see the result. Currently I am study the logs of wine in testruns but can not figure out whats wrong or what they doing when they launch on an OS higher than 8.1

**Comment by urusai88** [comment]
Well, from today the old interface is no longer available. When launched with Windows 7, an error pops up. When launched with Windows 10 or 11, a black window opens through which it is impossible to launch games.

**Comment by gegy1** [comment]
Thats bad news. Nevertheless, i am not going to buy any game from ubisoft until they remove this stupid launcher.

**Comment by urusai88** [comment]
I found a temporary solution
1) Create a new bottle
2) Install dotnet48
3) Install ubisoft connect launcher, DO NOT LAUNCH it after installation
4) Go to the root directory of the launcher, open the version.txt file, change the version to 11552
5) The old version of the launcher will not be updated to the new one. For now

**Comment by phistep** [comment]
> I found a temporary solution
> 
> 1. Create a new bottle
> 2. Install dotnet48
> 3. Install ubisoft connect launcher, DO NOT LAUNCH it after installation
> 4. Go to the root directory of the launcher, open the version.txt file, change the version to 11552
> 5. The old version of the launcher will not be updated to the new one. For now

Create a _Windows 10_ bottle :)

**Comment by gegy1** [comment]
> I found a temporary solution
> 
> 1. Create a new bottle
> 2. Install dotnet48
> 3. Install ubisoft connect launcher, DO NOT LAUNCH it after installation
> 4. Go to the root directory of the launcher, open the version.txt file, change the version to 11552
> 5. The old version of the launcher will not be updated to the new one. For now

Yes, but i need the new one, because of compatibility to online play. So now i need to find a way to run the new launcher. ;)

**Comment by gegy1** [comment]
As I mentioned, Anno 1800 does work but multiplayer doesn't. 
I found this:
https://github.com/ValveSoftware/wine/pull/173

**Comment by gegy1** [comment]
I think „Missing Kerberos support“ is the elephant in the room.
